### ISSUE TITLE
Pushing from rootNavCtrl 

### ISSUE THREAD

**Issue by pablogarciamiranda** [zyra/ionic2-super-tabs#273]
Hello!

I am trying to pushing a page inside SuperTabs and my desired behavior would be to hide SuperTabs  and show my tabs from below. From here I am doing well as show in the next images:

![image](https://user-images.githubusercontent.com/9169382/36790053-da87e8c0-1c93-11e8-8b10-72a429690b27.png)

![image](https://user-images.githubusercontent.com/9169382/36790024-bcefd82c-1c93-11e8-967e-9021e8fb8b1e.png)

But then, when I pop back to the SuperTabs page the SuperTabs disappear.

![image](https://user-images.githubusercontent.com/9169382/36789963-8ca3f46e-1c93-11e8-82e3-1c4664b14dd5.png)

I am doing it in the following way:

```
import {Component, Input} from '@angular/core';
import {App, NavParams} from 'ionic-angular';
import {StudentPage} from '../../../../pages/student/student';

@Component({
    selector: 'students-header',
    templateUrl: 'students-header.html'
})
export class StudentsHeaderComponent {

    onViewStudent(student) {
        this.navParams.get('rootNavCtrl').push(StudentPage, {student: student});
    }
}
```

I have tried to override the default back button action to pop back to the original SuperTabs page but it does not matter.

Is it a bug or am i doing something wrong?

Many thanks!

**Comment by eduardoRoth** [comment]
Hi @pablogarciamiranda,

Are `Parada 1` and `Parada 2` also SuperTabs or just a plain `ion-segment`?

For what I can see, there seems to be a problem with the container where you `StudentsPage` get loaded.

Try the following:

Replace this
```
   onViewStudent(student) {
        this.navParams.get('rootNavCtrl').push(StudentPage, {student: student});
    }
```

with this:
```
   onViewStudent(student){
      this.app.getRootNav().push(StudentPage, {student:student});
   }
```
_Don't forget to add app reference in your constructor `constructor(private app : App){}`_
_Also, getRootNav() is going to be deprecated in the next major release of Ionic, but there's an alternative_

What this does is that it gets the root navigation controller of your app and then pushes your page from there, after that, when you tap on "Back" the page will be popped from the root navigation controller and show the tab which you were.

**Comment by eduardoRoth** [comment]
@pablogarciamiranda 

So, are you using two SuperTabs components? One at top and the other at bottom?

**Comment by pablogarciamiranda** [comment]
@eduardoRoth No, the ones from the bottom are the tabs provided for Ionic. That tabs provide a global navigation through the app. 

And then, in one of that tabs I have a SuperTabs component.

Once I click on onViewStudent, which is inside SuperTabs, the tabs from the bottom must appear. But if I do what you are proposing me, they are not showing up and also when I come back to the page where the SuperTabs are placed, the tabs from top also disappeared (although they are swipeable)

**Comment by eduardoRoth** [comment]
@pablogarciamiranda 
Ok, what happens when you just inject NavController in your Paradas page and push your StudenPage from there?

**Comment by pablogarciamiranda** [comment]
@eduardoRoth 
Paradas page is where the SuperTabs is placed, so it won't have to much sense to push a page from there, am I right? I think you mean to push it from one page of SuperTabs.

This is how it looks.

![image](https://user-images.githubusercontent.com/9169382/36996381-02d90c74-20b7-11e8-850d-11e867c653b0.png)

And then, when coming back to SuperTabs page again they disappear.

And sure, here is part of my code:

**stops.html**. It contains the SuperTabs component.
  ```
  <super-tabs scrollTabs="true" *ngIf="stops">
        <super-tab color="navbar" *ngFor="let stop of stops; let i = index" [root]="stopPage" [rootParams]="stop"
                   title="Parada {{i+1}}"></super-tab>
  </super-tabs>
```
**stops.ts**
```
@Component({
    selector: 'page-stops',
    templateUrl: 'stops.html'
})
export class StopsPage {

    stops: any[];
    stopPage: any = StopPage;

    ionViewWillEnter() {
        this.storage.get('tripId').then(tripId => {
            if (!this.stops) {
                this.studentsService.getStudents(tripId).subscribe(students => {
                    this.tripId = tripId;
                    this.stops = this.studentsService.groupStudentsByStop(students);
                })
            }
        });
    }
}
```

**tabs.html.** It contains my tab2Root, which is the tab from below Paradas.
```
<ion-tabs>
    ...
    <ion-tab [root]="stopsPage" tabTitle="Paradas" tabIcon="information-circle"></ion-tab>
</ion-tabs>
```

```
import {Component} from '@angular/core';
import {StopsPage} from "../stops/stops";
import {StudentsPage} from '../students/students';

@Component({
    templateUrl: 'tabs.html'
})
export class TabsPage {

    studentsPage = StudentsPage;
    stopsPage = StopsPage;

    constructor() {

    }
}
```

**students-header.ts**: It's a component which belongs to stopPage referenced in **stops.ts**. 

```
import {Component, Input} from '@angular/core';
import {App, NavController, NavParams} from 'ionic-angular';
import {StudentPage} from '../../../../pages/student/student';

@Component({
    selector: 'students-header',
    templateUrl: 'students-header.html'
})
export class StudentsHeaderComponent {

    @Input() student: any;

    constructor(private app: App, private navParams: NavParams, private navCtrl: NavController) {
    }

    onViewStudent(student) {
        //this.navParams.get('rootNavCtrl').push(StudentPage, {student: student});
        this.navCtrl.push(StudentPage, {student: student});
    }
}

```

**Comment by pablogarciamiranda** [comment]
@eduardoRoth 
Thanks for your reply! The first approach also doesn't work. But I think I've found the error. 

A hidden directive which apply a display none property is being applied when coming back. A quick fix would be to override that behavior with a display block, but I don't figure out how to override it.

I am doing it in the following way, but the display none is still over it:

```
super-tabs-toolbar {
      display: block !important;
}
```
I am thinking about modifying the super-tabs-toolbar behavior, but i would like to know if there is another approach.

**Comment by saihanuman73** [comment]
@pablogarciamiranda Hi, Is your issue resolved ??

**Comment by senturkzuhtu11** [comment]
video link?

**Comment by egave** [comment]
Hi, I have the same issue with ionic2-super-tabs: ^4.1.3
Did you find a way to make your super-tabs display when poping back form another page ?
Regards

**Comment by pablogarciamiranda** [comment]
As quick fix, set scrollTabs to false. It should work @egave @saihanuman73

**Comment by egave** [comment]
Hi @pablogarciamiranda ,
thank you for your reply.
Unfortunately your quick fix didn't worked for me. 
As a workaround, I overrode the default back button action to make a "setRoot" instead of a "pop". Doing so my "super-tabs" are displayed. 

```
this.navBar.backButtonClick = (ev:UIEvent) =>
      this.navCtrl.setRoot(pageToGoBackTo);
    }
```

**Comment by egave** [comment]
Hi @pablogarciamiranda, @saihanuman73 
finally found a definitive solution. Upgrading to ionic-angular@3.9.2 solved the issue as the problem seems to have been solved in 3.7.1-201710201922 as told in theses thread.
[https://github.com/ionic-team/ionic/issues/13028](url)
[https://github.com/ionic-team/ionic/issues/13172](url)
